library.js

```diff
@@ -11,55 +12,66 @@ class Book {
   }
 }
 
-const mybooks = JSON.parse(localStorage.getItem('bookArchive')) || [];
+class Libraryshelf {
+  constructor() {
+    this.booksrecord = JSON.parse(localStorage.getItem('bookArchive')) || [];
+  }
 
-function displayBook(title, author, id) {
-  const templateHTML = `
-    ${title}<br>
-    ${author}<br>
-    <button type='button' class="remove" id="${id}">Remove</button>
-    <hr>`;
+  static displayBook(title, author, id) {
+    const templateHTML = `
+      <tr> 
+        <td class="table-item"><p>"${title}" by ${author}</p>
+        <button type='button' class="remove" id="${id}">Remove</button>
+        </td>
+      </tr>
+      `;
 
-  allbooks.insertAdjacentHTML('beforeend', templateHTML);
-}
+    allbooks.insertAdjacentHTML('beforeend', templateHTML);
+  }
 
-function getBooks() {
-  const storage = JSON.parse(localStorage.getItem('bookArchive'));
-  storage.forEach((book) => {
-    displayBook(book.title, book.author, book.id);
-  });
-}
+  static getBooks() {
+    const storage = JSON.parse(localStorage.getItem('bookArchive'));
+    storage.forEach((book) => {
+      Libraryshelf.displayBook(book.title, book.author, book.id);
+    });
+  }
 
-function addBook() {
-  libraryForm.addEventListener('submit', () => {
-    const inputBook = new Book(title.value, authors.value);
-    mybooks.push(inputBook);
-    title.value = '';
-    authors.value = '';
-    title.focus();
-    authors.focus();
-    localStorage.setItem('bookArchive', JSON.stringify(mybooks));
-    getBooks();
-  });
+  remove(button, key) {
+    button.addEventListener('click', () => {
+      if (key === 0) {
+        this.booksrecord.splice(key, key + 1);
+      } else {
+        this.booksrecord.splice(key, 1);
```

Comment:
- [**Optional**] The **splice()** methods mutate an array and other methods like **pop()**, **push()**, **sort()** etc. It will be nice to stick to methods that don't mutate an array such as **filter()**, **slice()**, **concat()**, **map()**.